### ISSUE TITLE
Cannot change/remove primary contact of place

### ISSUE THREAD

**Issue by abbyad** [medic/medic-webapp#3889]
In 2.13.0-beta.13, it is not possible to change or remove the primary contact of a place. Disassociating the primary contact from a place is necessary to delete the place. Changing the primary contact also happens.

To reproduce:
1) Create a person
1) Create a place, with the person from step above as primary contact
1) Edit place, and **remove the primary contact** by clicking the `x` in the select2 and then saving
1) Notice that the primary contact on the profile (and doc) is still the original contact
1) Edit place, and **change the primary contact** to another person then save
1) Notice that the primary contact on the profile (and doc) is still the original contact

**Comment by garethbowen** [comment]
@abbyad Works for me. When creating an issue please include the instance, username, and form used to reproduce this issue.

**Comment by abbyad** [comment]
Any user on standard.dev

**Comment by abbyad** [comment]
I'll add more info when I get back to a computer in a couple hours

**Comment by alxndrsn** [comment]
Confirmed on `master`.

**Comment by alxndrsn** [comment]
## ANTONIO:

<img width="483" alt="screen shot 2017-09-12 at 16 26 27" src="https://user-images.githubusercontent.com/191496/30334326-2cfb6a7e-97d7-11e7-9cc6-66418b0ca784.png">

## PLACE BEFORE:

<img width="672" alt="screen shot 2017-09-12 at 16 24 59" src="https://user-images.githubusercontent.com/191496/30334324-2cf6b63c-97d7-11e7-889a-7b5d689c0ae3.png">

## FORM DATA:

<img width="810" alt="screen shot 2017-09-12 at 16 25 30" src="https://user-images.githubusercontent.com/191496/30334323-2cf67c4e-97d7-11e7-8e7d-b66d2dda1700.png">

## PLACE AFTER:

<img width="742" alt="screen shot 2017-09-12 at 16 26 01" src="https://user-images.githubusercontent.com/191496/30334325-2cf8df70-97d7-11e7-9c6b-a2ea76898184.png">

# Conclusion

`<select2>` isn't updating the ID in the model when its value is changed.

**Comment by alxndrsn** [comment]
It looks like this is because there's no field called `/data/init/_id`.

I'm slightly confused why the `<init>` element is used to store intermediate values - the name implies that it's used for initial values rather than ongoing calculations.

**Comment by garethbowen** [comment]
@abbyad The form definition looks overly complicated to me - the dropdown binds to `/data/init/select_person` which is then updates `/data/health_center/contact`. Why doesn't the dropdown just update `/data/health_center/contact` directly?

**Comment by abbyad** [comment]
Adding a `data/init/_id` to the model didn't seem to help. Not to blame this on #2635, but it is possible that it is handled and stored differently now. I tried updating the form to set `/data/clinic/contact/_id` instead of `/data/clinic/contact`, but that didn't work. Any other ideas for form changes @alxndrsn?

Also, the reason it is `init` is likely because this was copied from a form where the contact info was used as initial values. I suspect we could safely update to `/data/contact` to better reflect how it is used, especially as this is a Standard contact form.

**Comment by garethbowen** [comment]
I can reproduce it with that form too, but not with other similar forms. My current theory is it's something to do with the circular calculations between `/data/init/select_person` and `/data/health_center/contact`. I'll keep digging later on tonight.

**Comment by abbyad** [comment]
> Why doesn't the dropdown just update /data/health_center/contact directly?

We use some of the contact's fields, like `name` in the form, so if I recall correctly we brought the select2 out of the place's doc so that `/data/{place}/contact` is just the id, and not all the contact's fields.

**Comment by alxndrsn** [comment]
Is it related that the new contact does not belong to a place?  Or at least, he appears to belong to the _wrong_ place:

<img width="378" alt="screen shot 2017-09-13 at 17 55 59" src="https://user-images.githubusercontent.com/191496/30387486-ee8486be-98ac-11e7-9409-776ae9e646fc.png">
<img width="439" alt="screen shot 2017-09-13 at 17 56 05" src="https://user-images.githubusercontent.com/191496/30387485-ee81e81e-98ac-11e7-9d04-1c45c02d56f1.png">


<img width="452" alt="screen shot 2017-09-13 at 17 56 28" src="https://user-images.githubusercontent.com/191496/30387484-ee80612e-98ac-11e7-89d4-217b69069756.png">
<img width="398" alt="screen shot 2017-09-13 at 17 56 25" src="https://user-images.githubusercontent.com/191496/30387487-ee8e9c08-98ac-11e7-957c-b5621cc2736b.png">

**Comment by garethbowen** [comment]
Changing the primary contact works if I remove the `calculate="../../health_center/contact"` from this binding: `<bind calculate="../../health_center/contact" nodeset="/data/init/select_person" type="db:person"/>`. Obviously the field then doesn't default to the current contact so it's not a solution but it's an extra piece of information...

**Comment by garethbowen** [comment]
Code review please @alxndrsn 

This change means this form no longer needs a circular reference - the name can be bound into the `health_center/contact/name` which is discarded on submission.

This is the diff from the [existing form in standard](https://github.com/medic/medic-projects/blob/master/standard/forms/contact/health_center-edit.xml) to the form I used for development that uses this new feature.

```
83c83
<             <value><output value=" /data/health_center/contact/name "/>'s Area</value></text>
---
>             <value><output value=" /data/init/contact_name "/>'s Area</value></text>
120c120
<             <value><output value=" /data/health_center/contact/name "/>'s Health Center</value></text>
---
>             <value><output value=" /data/init/contact_name "/>'s Health Center</value></text>
140c140
<             <value><output value=" /data/health_center/contact/name "/>'s District</value></text>
---
>             <value><output value=" /data/init/contact_name "/>'s District</value></text>
227c227
<             <value><output value=" /data/health_center/contact/name "/> का क्षेत्र</value></text>
---
>             <value><output value=" /data/init/contact_name "/> का क्षेत्र</value></text>
265c265
<             <value><output value=" /data/health_center/contact/name "/> का स्वास्थ्य केंद्र</value></text>
---
>             <value><output value=" /data/init/contact_name "/> का स्वास्थ्य केंद्र</value></text>
285c285
<             <value><output value=" /data/health_center/contact/name "/> का ज़िला</value></text>
---
>             <value><output value=" /data/init/contact_name "/> का ज़िला</value></text>
371c371
<             <value>Eneo ya  <output value=" /data/health_center/contact/name "/></value></text>
---
>             <value>Eneo ya  <output value=" /data/init/contact_name "/></value></text>
408c408
<             <value>Kituo cha afya cha  <output value=" /data/health_center/contact/name "/></value></text>
---
>             <value>Kituo cha afya cha  <output value=" /data/init/contact_name "/></value></text>
428c428
<             <value>WIlaya ya  <output value=" /data/health_center/contact/name "/></value></text>
---
>             <value>WIlaya ya  <output value=" /data/init/contact_name "/></value></text>
514c514
<             <value>Daerah <output value=" /data/health_center/contact/name "/></value></text>
---
>             <value>Daerah <output value=" /data/init/contact_name "/></value></text>
551c551
<             <value>Fasilitas Kesehatan <output value=" /data/health_center/contact/name "/></value></text>
---
>             <value>Fasilitas Kesehatan <output value=" /data/init/contact_name "/></value></text>
571c571
<             <value>Kabupaten <output value=" /data/health_center/contact/name "/></value></text>
---
>             <value>Kabupaten <output value=" /data/init/contact_name "/></value></text>
658c658
<             <value><output value=" /data/health_center/contact/name "/>को क्षेत्र</value></text>
---
>             <value><output value=" /data/init/contact_name "/>को क्षेत्र</value></text>
696c696
<             <value><output value=" /data/health_center/contact/name "/>को स्वास्थ्य केन्द्र</value></text>
---
>             <value><output value=" /data/init/contact_name "/>को स्वास्थ्य केन्द्र</value></text>
716c716
<             <value><output value=" /data/health_center/contact/name "/>को जिल्ला</value></text>
---
>             <value><output value=" /data/init/contact_name "/>को जिल्ला</value></text>
754,757c754
<             <contact>
<               <_id/>
<               <name/>
<             </contact>
---
>             <contact/>
791a789,790
>       <bind calculate="../name" nodeset="/data/init/contact_name" type="string"/>
>       <bind calculate="../../health_center/contact" nodeset="/data/init/select_person" type="db:person"/>
801,802c800
<       <bind nodeset="/data/health_center/contact/_id" type="db:person"/>
<       <bind calculate="." nodeset="/data/health_center/contact/name" type="string"/>
---
>       <bind calculate="coalesce( /data/init/select_person , .)" nodeset="/data/health_center/contact" type="string"/>
861c859
<       <input appearance="db-object" ref="/data/health_center/contact/_id">
---
>       <input appearance="db-object" ref="/data/init/select_person">
```

**Comment by garethbowen** [comment]
Merged and backported to `2.13.x`

**Comment by abbyad** [comment]
I have made changes to the place-edit forms for Standard config, and this is working well in `2.13.0-beta.16`. Moving to Ready, but @sglangevin we'll need to modify the contact forms for existing projects when the upgrade to ensure that they work with this change.
